File: Backend/AssemblyManager.cpp
#include "AssemblyManager.h"
#include "Conversion.h"
#include <iostream>
using namespace System;
using namespace System::Reflection;
using namespace System::IO;
using namespace System::Collections::Generic;
namespace Zuba {
	array<System::Reflection::Assembly^>^ AssemblyManager::GetAssemblies() {
        array<System::String^>^ dlls = File::ReadAllLines(".dllsheet");
        List<Assembly^>^ assemblies = gcnew List<Assembly^>();
        String^ loc = Assembly::GetExecutingAssembly()->Location;
        loc = Path::GetDirectoryName(loc);
        for each(String^ dll in dlls)
        {
            String^ path = loc + L'\\' + dll;
            assemblies->Add(Assembly::LoadFile(path));
        }
        return assemblies->ToArray();
	}
}

File: Zuba/AssemblyManager.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace Zuba
{
    internal static class AssemblyManager
    {
        internal static Assembly[] GetAssemblies()
        {
            string[] dlls = File.ReadAllLines(".dllsheet");
            List<Assembly> assemblies = new List<Assembly>();
            string loc = Assembly.GetExecutingAssembly().Location;
            loc = Path.GetDirectoryName(loc);
            foreach (string dll in dlls)
            {
                string path = loc + '\\' + dll;
                assemblies.Add(Assembly.LoadFile(loc + '\\' + dll));
            }
            return assemblies.ToArray();
        }
    }
}
